### PR TITLE
[ingress-nginx] update nginx tpl for healthz handle

### LINKED ISSUE #5134: Ingress-controller external healthcheck returns code 200 when pod is in terminating state
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v1.46.10

### Expected Behavior

Ingress-controller healthcheck must **NOT** return code 200 when pod is in `Terminating` state

### Actual Behavior

We have external balancer in CloudFlare, which checks `:80/healthz` on nodes with ingress-controller with `hostPort`. During ingress-controller update its pods turn into `Terminating` state, but `:80/healthz` continues to return code 200 OK. It leads that external loadbalancer continues to send requests to ingress, which it already cannot serve being in `Terminating` state.

### Steps To Reproduce

_No response_

### Additional Information

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by miklezzzz**
## Description
Changes:
- Update controller's `Check` method so as to check if a request has the `D8-External-Check` header set:
  a) if the header is set, `Check` takes into account the internal flag `isShuttingDown` value when calculating the response;
  b) if the header isn't set, `isShuttingDown` values is ignored (so as to not affect controller's readiness/liveness probes during Terminating state).

- nginx.tmpl was updated:
  a) **Breaking Change** - the server listening on :8080 was removed as unneeded. The LB's health checks using :8080/healthz require readjusting to :80/healthz.
  b) the default server `healthz` location was updated to set the `D8-External-Check` header and proxy pass connections to the controller's `healthcheck` handler (127.0.0.1:10254 in our case) so that accessing /healthz from external would return http.code in accordance with the current state of the controller.
Close #5134 

## Why do we need it, and what problem does it solve?
The `Check` method logic's been updated so that it can be used for both internal (liveness/readiness kubelet probes checking if the container is ready) and external (from load balancers checking if traffic should be withdrawn) checks.

The controller's healthz endpoint's been exposed via proxy_pass because it is binded to localhost(127.0.0.1:10254) and only /healthz location is exposed as the controller's  metrics are also served via the same port.

Unnecessary lifecycle hooks for `hostPort` and `hostPortWithProxyProtocol` inlets have been removed as a stale configuration.

## What is the expected result?
The expected result is that /healthz handle is available internally and externally. For internal checks `isShuttingDown` controller state should be ignored, but not for external ones.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: ingress-nginx
type: chore
summary: Fix `ingress-nginx` healthz handler replying with 200 unconditionally.
impact: All Ingress nginx controllers' pods will be recreated consecutively. Ingress nginx pods will stop to reply on :8080/healthz in favour of :80/healthz. Some LB's health checks might need readjusting.
impact_level: high
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by yalosev — DISMISSED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/402-ingress-nginx/docs/FAQ.md [hunk 1]
```diff
@@ -166,6 +166,15 @@ nginx.ingress.kubernetes.io/configuration-snippet: |
 {% endraw %}
 [Here](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#x509-client-certs) you can read more about how certificate authentication works.
 
+## How do I configure an external load balancer to check if IngressNginxController is available?
+
+In case an IngressNginxController is deployed behind a load balancer, it is advisable to configure your load balancer so that it would check
+the availability of the IngressNginxController's endpoints via a health check mechanism, periodically sending HTTP or TCP requests.
+While it is possible to test the endpoints simply by checking if a relevant TCP port is open, we recommend implementing HTTP checks with the following parameters:
+- Protocol: HTTP
+- Path: /healthz
```
Comment (z9r5) [inline modules/402-ingress-nginx/docs/FAQ.md:175]:
```suggestion
- Path: `/healthz`
```

File: modules/402-ingress-nginx/docs/FAQ.md [hunk 2]
```diff
@@ -166,6 +166,15 @@ nginx.ingress.kubernetes.io/configuration-snippet: |
 {% endraw %}
 [Here](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#x509-client-certs) you can read more about how certificate authentication works.
 
+## How do I configure an external load balancer to check if IngressNginxController is available?
+
+In case an IngressNginxController is deployed behind a load balancer, it is advisable to configure your load balancer so that it would check
+the availability of the IngressNginxController's endpoints via a health check mechanism, periodically sending HTTP or TCP requests.
+While it is possible to test the endpoints simply by checking if a relevant TCP port is open, we recommend implementing HTTP checks with the following parameters:
+- Protocol: HTTP
```
Comment (z9r5) [inline modules/402-ingress-nginx/docs/FAQ.md:174]:
```suggestion
- Protocol: `HTTP`
```

File: modules/402-ingress-nginx/docs/FAQ_RU.md [hunk 3]
```diff
@@ -167,6 +167,15 @@ nginx.ingress.kubernetes.io/configuration-snippet: |
...
 Подробнее о том, как работает аутентификация по сертификатам, можно прочитать в [документации Kubernetes](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#x509-client-certs).
 
+## Как сконфигурировать балансировщик нагрузки для проверки доступности IngressNginxController?
+
+В ситуации когда IngressNginxController размещен за балансировщиком нагрузки, рекомендуются сконфигурировать балансировщик для проверки доступности
+узлов IngressNginxController с помощью HTTP или TCP проверок. В то время как тестирование с помощью TCP пакетов представляет из себя простой и универсальный механизм проверки
+доступности, мы рекомендуем использовать проверку на основе HTTP-запросов со следующими параметрами:
+- Протокол: HTTP
+- Путь: /healthz
+- Порт: 80 (в случае использования inlet HostPort нужно указать номер порта, соответствующий [httpPort](cr.html#ingressnginxcontroller-v1-spec-hostport-httpport) настройке.
```
Comment (z9r5) [inline modules/402-ingress-nginx/docs/FAQ_RU.md:177]:
```suggestion
- Протокол: `HTTP`
- Путь: `/healthz`
- Порт: `80` (в случае использования inlet `HostPort` нужно указать номер порта, соответствующий параметру [httpPort](cr.html#ingressnginxcontroller-v1-spec-hostport-httpport).
```

**Review by yalosev — APPROVED** [review]

**Pull request merged**